### PR TITLE
Fix token & standard authentication not happening at same time during request

### PR DESCRIPTION
**Pull request by topcheese13**
Issue: https://github.com/vanilla/vanilla/issues/7950

After talking a little with both Ryan and Todd, the goal here was to trigger the token based authentication at the same "time" than standard auth to prevent different behaviours depending on your auth method as explained in the issue linked above. After taking both of your inputs into consideration this is what I came up with.

We add an event in the startAuthenticator method of the Auth class. We then change on what even the checkAccessToken method of the dashboard hooks is being called. Before `gdn_dispatcher_appStartup_handler` ,now `gdn_auth_startAuthenticator_handler`.

I have tested the changes in this PR and the API now works as expected. Let me know what you guys  think @initvector @tburry.

There is also a small casening fix in there.

### CODE REVIEW

File: applications/dashboard/settings/class.hooks.php [hunk 1]
```diff
@@ -663,7 +663,13 @@ public function gdn_dispatcher_appStartup_handler($sender) {
                 redirectTo($url);
             }
         }
+    }
+
+    public function gdn_auth_startAuthenticator_handler($sender) {
         $this->checkAccessToken();
+        if (Gdn::session()->isValid()) {
+            $sender->authHandled = true;
```
Comment (initvector) [inline applications/dashboard/settings/class.hooks.php]: This event property kinda seems redundant. Couldn't I check the result of `Gdn::session()->isValid()`, here and in other areas, to see if authentication had already been handled? It's a very lightweight method.

Comment (topcheese13) [reply applications/dashboard/settings/class.hooks.php]: Maybe I misunderstood this but when I proposed using the isValid method of the session, @tburry wanted a way to invalidate a previous auth if we need it in the future. He can probably provide deeper info.

Once this is settled I will definitely add a doc bloc for this.

File: applications/dashboard/settings/class.hooks.php [hunk 2]
```diff
@@ -663,7 +663,18 @@ public function gdn_dispatcher_appStartup_handler($sender) {
                 redirectTo($url);
             }
         }
+    }
+
+    /**
+     * Hooking on the startAuthenticator event of the auth class.
+     *
+     * @param $sender Auth
```
Comment (initvector) [inline applications/dashboard/settings/class.hooks.php]:
```suggestion
     * @param Gdn_Auth $sender
```

File: applications/dashboard/settings/class.hooks.php (hunk 1 above)
Comment (tburry) [reply applications/dashboard/settings/class.hooks.php]: I'm going to go back on what I said and agree with @initvector to make things simpler. We already have an improved way of doing this kind of thing in the future so we can just use `Gdn::session()->isValid()` on this for now. Sorry for the doublespeak.

**Pull request merged**